lib/fw.c

```diff
@@ -220,7 +222,7 @@ int switchtec_fw_write_fd(struct switchtec_dev *dev, int img_fd,
 
 		ret = switchtec_fw_wait(dev, &status);
 		if (ret != 0)
-		    return ret;
+			return status;
```

Comment:
This doesn't look correct. We might return the wrong thing when 'ret' actually is the error.

I'd suggest we change switchtec_fw_wait() to return a different value if the status indicates a CRC error.